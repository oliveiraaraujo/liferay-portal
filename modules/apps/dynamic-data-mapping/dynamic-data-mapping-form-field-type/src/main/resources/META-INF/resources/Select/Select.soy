{namespace Select}

{template .render}
	{@param name: string}
	{@param spritemap: string}
	{@param value: ?}
	{@param? _handleExpandedChanged: any}
	{@param? _handleItemClicked: any}
	{@param? _handleLabelClosed: any}
	{@param? dataSourceType: string}
	{@param? displayErrors: bool}
	{@param? emptyOptionLabel: string}
	{@param? errorMessage: string}
	{@param? expanded: bool}
	{@param? id: string}
	{@param? label: string}
	{@param? localizedValue: ?}
	{@param? multiple: bool}
	{@param? options: list<?>}
	{@param? predefinedValue: ?}
	{@param? readOnly: bool}
	{@param? repeatable: bool}
	{@param? required: bool}
	{@param? showLabel: bool}
	{@param? tip: string}
	{@param? valid: bool}
	{@param? visible: bool}

	{let $displayValues: length($value) > 0 ? $value : $predefinedValue and length($predefinedValue) > 0 ? $predefinedValue : [] /}

	{call FieldBase.render}
		{param contentRenderer kind="html"}
			{call .content}
				{param _handleExpandedChanged: $_handleExpandedChanged /}
				{param _handleItemClicked: $_handleItemClicked /}
				{param _handleLabelClosed: $_handleLabelClosed /}
				{param dataSourceType: $dataSourceType /}
				{param displayValues: $displayValues /}
				{param emptyOptionLabel: $emptyOptionLabel /}
				{param expanded: $expanded /}
				{param id: $id /}
				{param multiple: $multiple /}
				{param name: $name /}
				{param options: $options /}
				{param readOnly: $readOnly /}
				{param spritemap: $spritemap /}
				{param value: $value /}
			{/call}
		{/param}
		{param displayErrors: $displayErrors /}
		{param errorMessage: $errorMessage /}
		{param label: $label /}
		{param localizedValue: $localizedValue /}
		{param name: $name /}
		{param readOnly: $readOnly /}
		{param ref: 'base' /}
		{param repeatable: $repeatable /}
		{param required: $required /}
		{param showLabel: $showLabel /}
		{param spritemap: $spritemap /}
		{param tip: $tip /}
		{param valid: $valid /}
		{param visible: $visible /}
	{/call}
{/template}

{template .content}
	{@param displayValues: ?}
	{@param name: string}
	{@param spritemap: string}
	{@param value: list<string>}
	{@param? _handleExpandedChanged: any}
	{@param? _handleItemClicked: any}
	{@param? _handleLabelClosed: any}
	{@param? emptyOptionLabel: string}
	{@param? expanded: bool}
	{@param? id: string}
	{@param? multiple: bool}
	{@param? options: list<?>}
	{@param? readOnly: bool}

	{if $options}
		{let $trigger kind="html"}
			{if not $readOnly}
				{call .hidden_select}
					{param displayValues: $displayValues /}
					{param multiple: $multiple /}
					{param name: $name /}
					{param options: $options /}
				{/call}
			{/if}

			<div class="form-builder-select-field input-group-container">
				{let $classes kind="text"}
					form-control results-chosen select-field-trigger

					{if $multiple}
						{sp}multiple-label-list
					{/if}
				{/let}

				{let $attributes kind="attributes"}
					class="{$classes}"

					{if $id}
						id="{$id}"
					{/if}

					{if $readOnly}
						disabled
					{/if}
				{/let}

				<ul {$attributes}>
					{foreach $currentValue in $value}
						{foreach $option in $options}
							{if $currentValue == $option.value}
								{if $multiple}
									<li>
										{call ClayLabel.render}
											{param closeable: true /}
											{param elementClasses: 'ddm-select-option-label' /}
											{param data: [
												'value': $option.value
											] /}
											{param events: [
												'close': $_handleLabelClosed
											]/}
											{param label: $option.label /}
											{param spritemap: $spritemap /}
											{param truncate: true /}
										{/call}
									</li>
								{else}
									{call SingleSelectedOption.render}
										{param label: $option.label /}
									{/call}
								{/if}
							{/if}
						{/foreach}

						{ifempty}
							<div class="option-selected option-selected-placeholder">
								{if $emptyOptionLabel != null}
									{$emptyOptionLabel}
								{/if}
							</div>
					{/foreach}

					<a class="select-arrow-down-container">
						{if $spritemap}
							{call ClayIcon.render}
								{param spritemap: $spritemap /}
								{param symbol: 'caret-double' /}
							{/call}
						{/if}
					</a>
				</ul>
			</div>
		{/let}

		{call ClayDropdownBase.render}
			{param disabled: $readOnly /}
			{param events: [
				'expandedChanged': $_handleExpandedChanged,
				'itemClicked': $_handleItemClicked
			]/}
			{param elementClasses: 'ddm-btn-full ddm-select-dropdown' /}
			{param expanded: $expanded /}
			{param items: $options /}
			{param label: $trigger /}
			{param ref: 'dropdown' /}
			{param searchable: length($options) > 6 ? true : false /}
			{param spritemap: $spritemap /}
			{param useSearchableWithForm: false /}
		{/call}
	{/if}
{/template}

/**
 * Prints hidden select tag.
 */
{template .hidden_select}
	{@param displayValues: ?}
	{@param name: string}
	{@param options: list<?>}
	{@param? multiple: bool}

	<select aria-label="select" class="form-control hide" id="{$name}" name="{$name}" {if $multiple}multiple size="{length($options)}"{/if}>
		{foreach $currentValue in $displayValues}
			<option selected="selected" value="{$currentValue}">{$currentValue}</option>
		{/foreach}
	</select>
{/template}