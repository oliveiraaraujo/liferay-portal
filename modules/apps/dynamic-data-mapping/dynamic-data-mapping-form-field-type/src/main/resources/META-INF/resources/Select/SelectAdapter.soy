{namespace SelectAdapter}

/**
 * SelectAdapter
 */
{template .render}
	<div key="ddm-field-type-select"></div>
{/template}