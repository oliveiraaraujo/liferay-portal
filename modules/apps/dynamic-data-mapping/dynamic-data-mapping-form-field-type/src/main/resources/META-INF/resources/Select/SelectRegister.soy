{namespace SelectRegister}

{deltemplate PageRenderer.RegisterFieldType variant="'select'"}
	{call SelectAdapter.render data="all" /}
{/deltemplate}