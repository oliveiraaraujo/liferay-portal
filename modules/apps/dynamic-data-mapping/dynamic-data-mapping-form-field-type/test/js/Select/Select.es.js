/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import '@testing-library/jest-dom/extend-expect';
import {fireEvent, getByTestId, waitForElement} from '@testing-library/react';
import ReactDOM from 'react';

import Select from '../../../src/main/resources/META-INF/resources/Select/Select.es';
import withContextMock from '../__mocks__/withContextMock.es';

let component;
const spritemap = 'icons.svg';

const SelectWithContextMock = withContextMock(Select);

describe('Select', () => {
	beforeAll(() => {
		ReactDOM.createPortal = jest.fn(element => {
			return element;
		});
	});

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('is not editable', () => {
		component = new SelectWithContextMock({
			readOnly: true,
			spritemap,
		});

		const dropdownTrigger = document.body.querySelector(
			'.select-field-trigger'
		);

		expect(dropdownTrigger).toHaveClass('disabled');
	});

	it('has a help text', () => {
		component = new Select({
			spritemap,
			tip: 'Type something',
		});

		expect(component).toMatchSnapshot();
	});

	it('has an id', () => {
		component = new Select({
			id: 'ID',
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('renders options', () => {
		component = new Select({
			options: [
				{
					checked: false,
					disabled: false,
					id: 'id',
					inline: false,
					label: 'label',
					name: 'name',
					showLabel: true,
					value: 'item',
				},
			],
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('renders no options when options come empty', () => {
		component = new Select({
			options: [],
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('has a label', () => {
		component = new Select({
			label: 'label',
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('is closed by default', () => {
		component = new Select({
			open: false,
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it("has class dropdown-opened when it's opened", () => {
		component = new Select({
			open: true,
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('has a placeholder', () => {
		component = new Select({
			placeholder: 'Placeholder',
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('has a predefinedValue', () => {
		component = new Select({
			predefinedValue: ['Select'],
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('is not required', () => {
		component = new Select({
			required: false,
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('puts an asterisk when field is required', () => {
		component = new Select({
			label: 'This is the label',
			required: true,
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('renders Label if showLabel is true', () => {
		component = new Select({
			label: 'text',
			showLabel: true,
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('has a spritemap', () => {
		component = new Select({
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('has a value', () => {
		component = new Select({
			spritemap,
			value: ['value'],
		});

		expect(component).toMatchSnapshot();
	});

	it('has a key', () => {
		component = new Select({
			key: 'key',
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('emits a field edit event when an item is selected', async () => {
		const handleFieldEdited = data => {
			expect(data).toEqual(
				expect.objectContaining({
					fieldInstance: expect.any(Object),
					originalEvent: expect.any(Object),
					value: [''],
				})
			);
		};

		const events = {fieldEdited: handleFieldEdited};

		component = new SelectWithContextMock({
			dataSourceType: 'manual',
			events,
			options: [
				{
					label: 'label',
					value: 'item',
				},
				{
					label: 'label2',
					value: 'item2',
				},
			],
			spritemap,
		});

		const dropdownTrigger = component.element.querySelector(
			'.form-builder-select-field.input-group-container'
		);

		fireEvent.click(dropdownTrigger);

		const dropdownItem = await waitForElement(() =>
			getByTestId(document.body, 'dropdownItem-0')
		);

		fireEvent.click(dropdownItem);
	});

	it('emits a field edit event when an item is selected using multiselect', async () => {
		const handleFieldEdited = data => {
			expect(data).toEqual(
				expect.objectContaining({
					fieldInstance: expect.any(Object),
					originalEvent: expect.any(Object),
					value: ['item7'],
				})
			);
		};

		const events = {fieldEdited: handleFieldEdited};

		component = new SelectWithContextMock({
			dataSourceType: 'manual',
			events,
			multiple: true,
			options: [
				{
					label: 'label1',
					name: 'name1',
					value: 'item1',
				},
				{
					label: 'label2',
					name: 'name2',
					value: 'item2',
				},
				{
					label: 'label3',
					name: 'name3',
					value: 'item3',
				},
				{
					label: 'label4',
					name: 'name4',
					value: 'item4',
				},
				{
					label: 'label5',
					name: 'name5',
					value: 'item5',
				},
				{
					label: 'label6',
					name: 'name6',
					value: 'item6',
				},
				{
					label: 'label7',
					name: 'name7',
					value: 'item7',
				},
			],
			spritemap,
		});

		const dropdownTrigger = component.element.querySelector(
			'.form-builder-select-field.input-group-container'
		);

		fireEvent.click(dropdownTrigger);

		const labelItem = await waitForElement(() =>
			getByTestId(document.body, 'labelItem-item7')
		);

		fireEvent.click(labelItem);

		expect(component).toMatchSnapshot();
	});

	it('emits a field edit event when an item is selected using multiselect', async done => {
		const handleFieldEdited = data => {
			expect(data).toEqual(
				expect.objectContaining({
					fieldInstance: expect.any(Object),
					originalEvent: expect.any(Object),
					value: ['item7', 'item2'],
				})
			);

			done();
		};

		const events = {fieldEdited: handleFieldEdited};

		component = new SelectWithContextMock({
			dataSourceType: 'manual',
			events,
			multiple: true,
			options: [
				{
					label: 'label1',
					name: 'name1',
					value: 'item1',
				},
				{
					label: 'label2',
					name: 'name2',
					value: 'item2',
				},
				{
					label: 'label3',
					name: 'name3',
					value: 'item3',
				},
				{
					label: 'label4',
					name: 'name4',
					value: 'item4',
				},
				{
					label: 'label5',
					name: 'name5',
					value: 'item5',
				},
				{
					label: 'label6',
					name: 'name6',
					value: 'item6',
				},
				{
					label: 'label7',
					name: 'name7',
					value: 'item7',
				},
			],
			spritemap,
			value: ['item7', 'item3', 'item2'],
		});

		const dropdownTrigger = component.element.querySelector(
			'.form-builder-select-field.input-group-container'
		);

		fireEvent.click(dropdownTrigger);

		const labelItemCloseButton = await waitForElement(() => {
			return getByTestId(dropdownTrigger, 'closeButtonitem3');
		});

		fireEvent.click(labelItemCloseButton);
	});

	it('renders the dropdown with search when there are more than six options', () => {
		component = new Select({
			dataSourceType: 'manual',
			options: [
				{
					label: 'label',
					name: 'name',
					value: 'item',
				},
				{
					label: 'label',
					name: 'name',
					value: 'item',
				},
				{
					label: 'label',
					name: 'name',
					value: 'item',
				},
				{
					label: 'label',
					name: 'name',
					value: 'item',
				},
				{
					label: 'label',
					name: 'name',
					value: 'item',
				},
				{
					label: 'label',
					name: 'name',
					value: 'item',
				},
				{
					label: 'label',
					name: 'name',
					value: 'item',
				},
			],
			spritemap,
		});

		expect(component).toMatchSnapshot();
	});
});
