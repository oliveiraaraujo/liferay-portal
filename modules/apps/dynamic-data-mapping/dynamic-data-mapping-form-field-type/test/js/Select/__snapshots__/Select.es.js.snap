// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select emits a field edit event when an item is selected using multiselect 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" size="7" class="form-control form-control">
      <option value="item7">label7</option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger multiple-label-list" tabindex="0">
        <li><span value="item7" class="label ddm-select-option-label label-dismissible label-secondary"><span class="label-item label-item-expand">label7</span><span class="label-item label-item-after"><button data-testid="closeButtonitem7" class="close" type="button"><svg class="lexicon-icon lexicon-icon-times" role="presentation">
                  <use xlink:href="icons.svg#times"></use>
                </svg></button></span></span></li><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select has a help text 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div><span class="form-text">Type something</span>
  </div>
</div>
`;

exports[`Select has a key 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select has a label 1`] = `
<div class="react-component-adapter">
  <div class="form-group">
    <p class="ddm-label">label</p><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select has a placeholder 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select has a predefinedValue 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select has a spritemap 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select has a value 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select has an id 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" id="ID" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select has class dropdown-opened when it's opened 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select is closed by default 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select is not required 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select puts an asterisk when field is required 1`] = `
<div class="react-component-adapter">
  <div class="form-group">
    <p class="ddm-label">This is the label<span class="reference-mark"><svg class="lexicon-icon lexicon-icon-asterisk" role="presentation">
          <use xlink:href="icons.svg#asterisk"></use>
        </svg></span></p><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders Label if showLabel is true 1`] = `
<div class="react-component-adapter">
  <div class="form-group">
    <p class="ddm-label">text</p><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders no options when options come empty 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders options 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders the dropdown with search when there are more than six options 1`] = `
<div class="react-component-adapter">
  <div class="form-group"><select multiple="" aria-label="select" hidden="" class="form-control form-control">
      <option disabled="" value=""></option>
    </select>
    <div class="form-builder-select-field input-group-container" style="">
      <div class="form-control results-chosen select-field-trigger" tabindex="0">
        <div class="option-selected option-selected-placeholder">choose-an-option</div><a class="select-arrow-down-container"><svg class="lexicon-icon lexicon-icon-caret-double" role="presentation">
            <use xlink:href="icons.svg#caret-double"></use>
          </svg></a>
      </div>
    </div>
  </div>
</div>
`;
